### PR TITLE
DOC: Tiny fixes, and possible overhaul, of the two scales example in the gallery

### PR DESCRIPTION
**Pull request by afvincent**
## PR Summary

Two-fold PR about the gallery example that shows how to use `ax.twinx` to display [two scales](https://matplotlib.org/devdocs/gallery/api/two_scales.html) on the same plot:
1. 367647e fixes a few typos and introduces a call to `fig.tight_layout` to prevent the y-labels from being clipped
2. fdb5900 is a significant style overhaul that:
    * compresses the docstring of the (main) helper function;
    * gets rid of the minor helper function used to color the y-tick labels, to simply include the relevant code into the main helper function (this avoids passing the desired colors a second time);
    * now colors the y-labels as well;
    * sets up the axes labels in a place that actually makes sense (beforehand, those labels were hard-coded into the main helper function, independently of whatever *data1* and *data2* could be...).

I agree that the value of the *2nd* commit may be subjective (although I personally think that the helper function is now quite more logical, and more suited to reusing outside of the gallery) and I would understand if it were discarded simply because of that. However, if it was the case, the *1st* commit fixes minor issues that, I think, are objective enough to still be this commit.

Here is what the example looks like with both commits:
with_both_commits

## PR Checklist

- [ ] Code is PEP 8 compliant: I think so, CI will decide
- [X] Documentation is sphinx and numpydoc compliant: AFAIU, it should be

NB: milestoned for 2.2 as it is only a small piece of documentation. Feel free to re-milestone for 3.0 if needed.

### CODE REVIEW

**Comment by timhoffm** [comment]
I would get rid of the main helper function as well. It just saves duplicating the four lines of plot code for the second axes, at the cost of a lot of docstring and loop over  a parameter set. As such it‘s distracting from the main point twinx.

Also, you wouldn‘t likely see such a function in real life. It’s too specific, except if you would do a lot of these plots, which is beyond the scope of this example.

**Comment by afvincent** [comment]
FWIW I had tried to keep some of the previous functional approach as I know that @tacaswell likes to promote it (IIRC). Anyway, a straight to the point example is fine with me too, so here it is.

(About the real-life plausibility, well I used such helper functions at my day-job ^^...)

File: examples/api/two_scales.py [hunk 1]
```diff
@@ -17,64 +17,26 @@
...
-    """Color your axes."""
-    for t in ax.get_yticklabels():
-        t.set_color(color)
-    return None
-color_y_axis(ax1, 'r')
-color_y_axis(ax2, 'b')
+data1 = np.exp(t)
+data2 = np.sin(2 * np.pi * t)
+
+# Create twin axes and plot the mock data onto them
```
Comment (QuLogic) [inline examples/api/two_scales.py]: This comment is kind of "small" for its contents. That is, it describes the next two or three blocks, but is just a regular comment. I feel like it should be broken up to each block, or made a bit more distinct visually.

**Comment by afvincent** [comment]
@QuLogic Is it better with the comments like in ce43564?

**Comment by timhoffm** [comment]
Wouldn't  it be more instructive for twinning, to plainly write everything out? E.g.

~~~
fig, ax1 = plt.subplots()
		
ax1.set_xlabel('time (s)')
ax1.set_ylabel('exp')
ax1.plot(t, data1, color='red')
ax1.yaxis.label.set_color('red')
ax1.tick_params(labelcolor='red')

ax2 = ax1.twinx()  # instantiate a second axes that shares the same x-axis

ax2.set_ylabel('sin') # NB: we already took care of the x-label with ax1
ax2.plot(t, data2, color='blue')
ax2.yaxis.label.set_color('blue')
ax2.tick_params(labelcolor='blue')

fig.tight_layout()  # otherwise the right y-label is slightly clipped
plt.show()
~~~

**Comment by dopplershift** [comment]
I like @timhoffm 's version better.

**Comment by afvincent** [comment]
I am not sure that I totally agree but I am too biased anyway 😈, so here is the unrolled version adapted from @timhoffm's suggestion.  

Being there, slight update of the color for something more in phase with Matplotlib 2+ :):
overhaul_of_the_overhaul

**Comment by afvincent** [comment]
Thanks all :)!

**Pull request merged**